lib/jekyll/filters/url_filters.rb

```diff
@@ -13,11 +13,16 @@ module URLFilters
       def absolute_url(input)
         return if input.nil?
         input = input.url if input.respond_to?(:url)
-        return input if Addressable::URI.parse(input.to_s).absolute?
+        input = input.to_s
+        return input if absolute?(input)
         site = @context.registers[:site]
         return relative_url(input) if site.config["url"].nil?
```

Comment:
IMO, this can be further optimized as well.. `relative_url(input)` will again trickle down to calling `input.nil?`, `input.respond_to?(:url)`, `absolute?(input)` before ending up inside `Addressable::URI.parse()`

Reply:
ping

Reply:
I agree that it could be further optimized. Feel free to push the commit to this branch; I might not have time for a few days.